cpp/include/functions.h

```diff
@@ -15,65 +15,189 @@
- * 
- * @Returns                       GDF_SUCCESS upon successful completion. 
+ *
+ * @Returns                       GDF_SUCCESS upon successful completion.
  */
 /* ----------------------------------------------------------------------------*/
 gdf_error gdf_pagerank(gdf_graph *graph, gdf_column *pagerank, float alpha, float tolerance, int max_iter, bool has_guess);
 
+/**
+ * @Synopsis   Creates source, destination and value columns based on rmat input
+ *
+ * @Param[in] *argv                  String that accepts the following arguments
+ *                                   rmat (default: rmat_scale = 10, a = 0.57, b = c = 0.19)
+ *                                               Generate R-MAT graph as input
+ *                                               --rmat_scale=<vertex-scale>
+ *                                               --rmat_nodes=<number-nodes>
+ *                                               --rmat_edgefactor=<edge-factor>
+ *                                               --rmat_edges=<number-edges>
+ *                                               --rmat_a=<factor> --rmat_b=<factor> --rmat_c=<factor>
+ *                                               --rmat_self_loops If this option is supplied, then self loops will be retained
+ *                                               --rmat_undirected If this option is not mentioned, then the graps will be undirected
+ *                                       Optional arguments:
+ *                                       [--device=<device_index>] Set GPU(s) for testing (Default: 0).
+ *                                       [--quiet]                 No output (unless --json is specified).
+ *                                       [--random_seed]           This will enable usage of random seed, else it will use same seed
+ *                                       [--normalized]\n
+ *
+ * @Param[out] &vertices             Number of vertices in the generated edge list
+ *
+ * @Param[out] &edges                Number of edges in the generated edge list
+ *
+ * @Param[out] *src                  Columns containing the sources
+ *
+ * @Param[out] *dst                  Columns containing the destinations
+ *
+ * @Param[out] *val                  Columns containing the edge weights
+ *
+ * @Returns                          GDF_SUCCESS upon successful completion.
+ */
+/* ----------------------------------------------------------------------------*/
 gdf_error gdf_grmat_gen (const char* argv, size_t &vertices, size_t &edges, gdf_column* src, gdf_column* dest, gdf_column* val);
 
+/**
+ * @Synopsis   Performs a breadth first search traversal of a graph starting from a node.
+ *
+ * @Param[in] *graph                 cuGRAPH graph descriptor with a valid edgeList or adjList
+ *
+ * @Param[out] *distances            If set to a valid column, this is populated by distance of every vertex is thr graph from the starting node
```

Comment:
"this is populated by distance of every vertex is thr graph from the starting node" -> need rework